unit/solvers/smt2_incremental/construct_value_expr_from_smt.cpp

```diff
@@ -103,14 +103,14 @@ TEST_CASE("Value expr construction from smt.", "[core][smt2_incremental]")
     rowt{smt_bit_vector_constant_termt{1, 8}, from_integer(1, c_bool_typet(8))},
     rowt{
       smt_bit_vector_constant_termt{0, 64},
-      from_integer(0, pointer_typet(void_type(), 64 /* bits */))},
+      from_integer(0, pointer_typet{empty_typet{}, 64 /* bits */})},
```

Comment:
My apologies for nitpicking something which is already merged, but how does swapping from `void_type()` to `empty_typet{}` fix a 32-bit build problem? The `void_type()` function just returns an instance of `empty_typet{}`, and neither of these depend on a bit width, as far as I can see.

Reply:
Oh, yes, that indeed wasn't a complete fix: it was my attempt to get rid of `util/c_types.h`, for many of the types in there are by definition platform-dependent. But turns out that fully removing that include is impossible, because there also is use of `c_bool_typet` and `c_enum_typet`.

Reply:
Ah, so I could effectively think of this as a front-end de-coupling effort. There are many C front end specific types in the unit tests for the new SMT decision procedure because the decision procedure needs to implement support for them specifically. This is the case because these data structures find there way all the way through cbmc to the decision procedure rather than being lowered to some kind of goto construct at the end of the processing of the front end. Another one for the list is `union_typet` which is defined in `util/c_types.h`, but which is sent to the decision procedure.

Whilst the decision procedure needs front-end specific functionality I would prefer to keep matching front-end specific types in the unit tests. That way we can remove the front end specifics from the decision procedure and the unit tests if and when we ever sufficiently decouple to stop them getting to the decision procedure.

Reply:
I entirely agree with your assessment. One day we should come clean on this and decide whether the types you mentioned are to be considered fundamental types, and therefore, should be moved out of `c_types.h`, or be lowered and not be ones to handle in back-ends.

Reply:
So, yes, my change in this file wasn't actually necessary, but probably doesn't need to be reverted either?

Reply:
I think it was somewhat easier to read as a void pointer, rather than an empty pointer. However I also don't think it is worth taking the time to revert. I questioned it as it didn't appear to match the commit message concerning 32 bit support, so I didn't initially understand why the change had been made. Thank you for taking the time to respond.